File: katas/6 kyu/split-strings/solution.py
# split-strings
import codewars_test as test
# -------------------------------- MY SOLUTION ------------------------------- #
def solution(s):
  return [s[i:i+2].ljust(2, "_") for i in range(0, len(s), 2)]
# ------------------------------ CLEVER SOLUTION ----------------------------- #
import re

def solution(s):
    return re.findall(".{2}", s + "_")
# ----------------------------------- TEST ----------------------------------- #
test.describe("Example Tests")

tests = (
    ("asdfadsf", ['as', 'df', 'ad', 'sf']),
    ("asdfads", ['as', 'df', 'ad', 's_']),
    ("", []),
    ("x", ["x_"]),
)

for inp, exp in tests:
    test.assert_equals(solution(inp), exp)

File: katas/6 kyu/split-strings/solution.php
<!-- split-strings -->
<?php
use PHPUnit\Framework\TestCase;
/* ------------------------------- MY SOLUTION ------------------------------ */
function solution($str) {
  if ($str == "") return [];
  $arr = str_split($str , 2);
  $temp = array_pop($arr);
  if (strlen($temp) < 2){
    $temp .= "_";
  }
  array_push($arr , $temp);
  return $arr;
}
/* ----------------------------- CLEVER SOLUTION ---------------------------- */
function solution($str) {
  if (empty($str))
    return [];
  if (strlen($str) % 2 != 0)
    $str .= "_";
  return str_split($str, 2);
}
/* ---------------------------------- TEST ---------------------------------- */
class MyTestCases extends TestCase
{
    public function testBasicTests() {
        $this->assertEquals(["ab", "cd", "ef"], solution("abcdef"));
        $this->assertEquals(["ab", "cd", "ef", "g_"], solution("abcdefg"));
        $this->assertEquals([], solution(""));
    }
}